### PR TITLE
ref: Remove option to pass dataset to consumer and replacer

### PR DESCRIPTION
**Pull request by lynnagara**
We should now only be passing a storage name and not a dataset name
to consumers and replacers.

Before merging this PR we will need to update the places where a
dataset is still specified:
- [x] ops 
- [x] onpremise

Single tenant will need to be updated only on the next release.

### CODE REVIEW

**Comment by fpacifici** [comment]
The dependencies on production changes seem fine. Though I am not sure it would work for single tenant. Is it possible that snuba image is updated without pulling the docker compose file from the onpremise repo? That would make it impossible to make this change.

**Comment by lynnagara** [comment]
SaaS and on premise (which both use latest Snuba master) have been updated now, so there are no more blockers to this PR. Single tenant runs an older version of Snuba so this will need to happen in line with the next upgrade. (ref: https://github.com/getsentry/single-tenant/pull/143)

File: snuba/cli/consumer.py [hunk 1]
```diff
@@ -154,15 +133,17 @@ def consumer(
...
+        ), "Only CDC storages have a control topic thus are supported."
         context = ConsumerStateMachine(
             consumer_builder=consumer_builder,
             topic=control_topic or storage.get_default_control_topic(),
             group_id=consumer_group,
             storage=storage,
         )
 
-        def handler(signum, frame) -> None:
+        def handler(signum: int, frame: Any) -> None:
```
Comment (fpacifici) [inline snuba/cli/consumer.py:146]: thanks

**Review by fpacifici — APPROVED** [review]
nice.  thanks.

**Pull request merged**